app/services/image_validation.py

```diff
@@ -0,0 +1,211 @@
+            raise HTTPException(
+                status_code=400,
+                detail='File content does not match declared type. Possible file corruption or spoofing.',
+            )
+
+        # Step 5: Validate image can be opened and get dimensions
+        try:
+            image = Image.open(io.BytesIO(file_content))
+            width, height = image.size
+            image_format = image.format  # PIL detected format
+
+            logger.info(f'Image opened successfully: {filename} ({width}x{height}, format: {image_format})')
+
+        except UnidentifiedImageError:
+            logger.warning(f'Cannot identify image format for {filename}')
+            raise HTTPException(status_code=400, detail='Invalid or corrupted image file.')
+        except Exception as e:
+            logger.error(f'Error opening image {filename}: {str(e)}')
+            raise HTTPException(status_code=400, detail='Cannot process image. File may be corrupted.')
+
+        # Step 6: Validate resolution
+        if width > settings.image_max_resolution or height > settings.image_max_resolution:
+            logger.warning(f'Resolution too high: {width}x{height} for file {filename}')
+            raise HTTPException(
+                status_code=400,
+                detail=f'Image resolution {width}x{height} exceeds maximum allowed {settings.image_max_resolution}x{settings.image_max_resolution}',
+            )
+
+        # Step 7: Reset file pointer for potential future reads
+        await file.seek(0)
+
+        # All validation passed - return metadata
+        logger.info(
+            f'Image validation successful: {filename} '
+            f'({width}x{height}, {file_size} bytes, {content_type})'
+        )
+
+        return {
+            'filename': filename,
+            'content_type': content_type,
+            'size_bytes': file_size,
+            'width': width,
+            'height': height,
+        }
+
+    except HTTPException:
+        # Re-raise validation errors
+        raise
+    except Exception as e:
+        # Catch any unexpected errors
+        logger.error(f'Unexpected error during image validation: {str(e)}', exc_info=True)
+        raise HTTPException(status_code=500, detail='An unexpected error occurred during image validation.')
+
+
+def _validate_magic_bytes(file_content: bytes, content_type: str) -> bool:
+    """
+    Validate file magic bytes match the declared content type.
+
+    This prevents file type spoofing by checking the actual file content
+    against known magic byte signatures.
+
+    Args:
+        file_content: Raw file bytes
+        content_type: Declared MIME type
+
+    Returns:
+        bool: True if magic bytes match content type, False otherwise
+    """
+    if content_type not in MAGIC_BYTES:
+        logger.warning(f'No magic bytes defined for content type: {content_type}')
+        return False
```

Comment:
Logging for missing magic bytes may be excessive for valid image types. Log only when file type is unsupported or suspicious.
```
if content_type not in MAGIC_BYTES:
    if content_type.startswith("image/"):
        logger.warning(f'Unknown image type: {content_type}')
    return False
```